File: app/src/main/java/org/cuieney/videolife/di/module/AppModule.java
package org.cuieney.videolife.di.module;

import android.content.Context;


import org.cuieney.videolife.App;
import org.cuieney.videolife.common.api.KyApiService;
import org.cuieney.videolife.common.api.OpApiService;
import org.cuieney.videolife.common.api.WyApiService;
import org.cuieney.videolife.di.RetrofitHelper;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;


@Module
public class AppModule {

    public final App mApp;


    public AppModule(App mApp) {
        this.mApp = mApp;
    }

    @Provides
    @Singleton
    Context providesContext() {
        return mApp;
    }


    @Provides
    KyApiService proviesKyService(RetrofitHelper retrofitHelper){
        return retrofitHelper.getKyApiService();
    }

    @Provides
    WyApiService proviesWyService(RetrofitHelper retrofitHelper){
        return retrofitHelper.getWyApiService();
    }


    @Provides
    OpApiService proviesOpService(RetrofitHelper retrofitHelper){
        return retrofitHelper.getOpApiService();
    }
}

File: app/src/main/java/org/cuieney/videolife/kotlin/di/module/AppModule.kt
package org.cuieney.videolife.kotlin.di.module

import android.content.Context
import dagger.Module
import dagger.Provides
import org.cuieney.videolife.common.api.*
import org.cuieney.videolife.kotlin.App
import org.cuieney.videolife.kotlin.di.RetrofitHelper
import javax.inject.Singleton


/**
 * Created by cuieney on 2017/5/22.
 */

@Module class AppModule(mApp: App){
    var app: App = mApp

    @Provides
    @Singleton
    fun providesContext(): Context = app

    @Provides
    @Singleton
    fun proviesKyService(retrofitHelper: RetrofitHelper): KyApiService {
        return retrofitHelper.getKyApiService()
    }

    @Provides
    @Singleton
    fun proviesWyService(retrofitHelper: RetrofitHelper): WyApiService {
        return retrofitHelper.getWyApiService()
    }

    @Provides
    @Singleton
    fun proviesOpService(retrofitHelper: RetrofitHelper): OpApiService {
        return retrofitHelper.getOpApiService()
    }

    @Provides
    @Singleton
    fun proviesVeerService(retrofitHelper: RetrofitHelper): VeerApiService {
        return retrofitHelper.getVeerApiService()
    }


    @Provides
    @Singleton
    fun proviesKuulaService(retrofitHelper: RetrofitHelper): KuulaApiService {
        return retrofitHelper.getKuulaApiService()
    }

}